src/RTCIceCandidate.ts

```diff
@@ -1,22 +1,12 @@
 export default class RTCIceCandidate {
     candidate: string;
-    sdpMLineIndex: number;
-    sdpMid: string;
+    sdpMLineIndex: number | null;
+    sdpMid: string | null;
 
-    constructor(info) {
-        if (typeof info?.candidate !== 'string') {
-            throw new TypeError('`candidate` must be string');
-        }
-        if (typeof info?.sdpMLineIndex !== 'number') {
-            throw new TypeError('`sdpMLineIndex` must be number');
-        }
-        if (typeof info?.sdpMid !== 'string') {
-            throw new TypeError('`sdpMid` must be string');
-        }
-
-        this.candidate = info.candidate;
-        this.sdpMLineIndex = info.sdpMLineIndex;
-        this.sdpMid = info.sdpMid;
+    constructor({ candidate = '', sdpMLineIndex = null, sdpMid = null }) {
```

Comment:
Can you make this throw in this case, as per the spec? https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidate/RTCIceCandidate#exceptions

Reply:
added